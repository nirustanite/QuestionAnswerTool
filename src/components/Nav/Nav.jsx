import React from 'react';
import { Menu, Container } from 'semantic-ui-react';
import { useRouteMatch, Link } from "react-router-dom";
import routes from 'Pages/routes';
import styled from 'styled-components';

const StyledMenu = styled(Menu)`
    border: none !important;

    .item {
        
        &:first-child {
            border-left: 0px !important;
        }

        &::before {
            display: none;
        }    

        &.active {
            box-shadow: 0px -2px 0px 0px #66CD00 inset !important;
            background: none !important;
        }
    }   
`;

const Nav = () => {

    const matchHome = useRouteMatch({
        path: "/",
        exact: true
    });

    return(
        <StyledMenu fixed="top" stackable>
            <Container>
                <Menu.Item>
                    QA Tool
                </Menu.Item>
                <Menu.Item 
                    as={Link}
                    name="HOME"
                    to={routes.HOME}
                    active={!!matchHome}
                >
                    Home
                </Menu.Item>
            </Container>
        </StyledMenu>
    );
};

export default Nav;