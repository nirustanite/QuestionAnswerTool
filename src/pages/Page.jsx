import React from "react";
import { Container } from 'semantic-ui-react';
import styled from 'styled-components';
import Nav from 'Components/Nav/Nav';

const StyledContainer = styled(Container)`
    padding-top: 5em;
    padding-bottom: 5em;
`;

const Page = ({ children }) => {
    return (
        <React.Fragment>
            <Nav />
            <StyledContainer>{children}</StyledContainer>
        </React.Fragment>
    );
};

export default Page;
