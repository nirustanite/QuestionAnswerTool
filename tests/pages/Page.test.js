import React from 'react';
import { shallow } from 'enzyme';
import Page from 'Pages/Page';
import Nav from 'Components/Nav/Nav';


describe('Page', () => {

    it('renders without error', () => {
		const wrapper = shallow(<Page />);
		expect(wrapper.length).toEqual(1);
	});

    it('should render the Page Component correctly', () => {   
        const child = "Hello"
        const wrapper = shallow(<Page>{child}</Page>);
        expect(wrapper).toMatchSnapshot();
    });

    it('should render a <Nav /> component', () => {
        const wrapper = shallow(<Page />);
        expect(wrapper.find(Nav)).toHaveLength(1);
    });

});
