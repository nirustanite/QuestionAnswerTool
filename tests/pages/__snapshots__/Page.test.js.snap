// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page should render the Page Component correctly 1`] = `ShallowWrapper {}`;
